src/dask_awkward/io.py

```diff
@@ -55,3 +52,36 @@ def from_parquet(source: Any, **kwargs: Any) -> DaskAwkwardArray:
 
     hlg = HighLevelGraph.from_collections(name, dsk)
     return new_array_object(hlg, name, None, npartitions=npartitions)
+
+
+def _chunked_to_json_v2(chunks):
+    return from_iter_v2(json.loads(ch) for ch in chunks.split(b"\n") if ch)
+
+
+def _chunked_to_json_v1(chunks):
+    return from_iter_v1(json.loads(ch) for ch in chunks.split(b"\n") if ch)
+
+
+def _from_json(
+    source: str | list[str],
+    concrete: Callable,
+    delimiter: bytes = b"\n",
+    blocksize: int | str = "128 MiB",
+) -> DaskAwkwardArray:
+    token = tokenize(source, delimiter, blocksize)
+    name = f"from-json-{token}"
+    _, chunks = read_bytes(
+        source,
+        delimiter=delimiter,
+        blocksize=blocksize,
+        sample=None,
+    )
+    chunks = list(flatten(chunks))
+    dsk = {(name, i): (concrete, d.key) for i, d in enumerate(chunks)}
+    hlg = HighLevelGraph.from_collections(name, dsk, dependencies=chunks)
```

Comment:
Once again, this looks awfully like a `Layer` that doesn't need to be materialised.

Reply:
Agreed!